src/scope/VariableGroup.js

```diff
@@ -41,9 +41,9 @@ class VariableGroup {
    *
    * @return {String} Either "var", "let", "const" or undefined.
    */
-  getCommonKind() {
-    const firstKind = this.variables[0].getKind();
-    if (this.variables.every(v => v.getKind() === firstKind)) {
+  getCommonKind(scope) {
+    const firstKind = this.variables[0].getKind(scope.findFunctionScoped(this.variables[0].node.id.name));
+    if (this.variables.every(v => v.getKind(scope.findFunctionScoped(v.node.id.name)) === firstKind)) {
```

Comment:
We're breaking the encapsulation of `Variable` class in here.

Instead of doing `v.node.id.name` we should introduce a new method to get a name of a variable, so we could just ask `v.getName()`. Really even for accessing the `node` we should have used the `getNode()` method.